includes/uninstall-functions.php

```diff
@@ -59,33 +59,19 @@ function delete_posts() {
 
 	global $wpdb;
 
-	$current_page = 0;
-	$per_page     = 1000;
-	$post_type    = 'amp_validated_url';
-
-	do {
-		$offset = $per_page * $current_page;
-
-		// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching -- Cannot cache result since we're deleting the records.
-		$post_ids = $wpdb->get_col(
-			$wpdb->prepare(
-				"SELECT ID FROM $wpdb->posts WHERE post_type = %s LIMIT %d OFFSET %d;",
-				$post_type,
-				$per_page,
-				$offset
-			)
-		);
-
-		if ( empty( $post_ids ) || ! is_array( $post_ids ) ) {
-			break;
-		}
+	// phpcs:disable WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching
 
-		foreach ( $post_ids as $post_id ) {
-			wp_delete_post( $post_id );
-		}
+	$wpdb->delete(
+		$wpdb->posts,
+		[
+			'post_type' => 'amp_validated_url',
+		]
+	);
 
-		$current_page++;
-	} while ( ! empty( $result ) );
+	// Delete orphan post meta data.
+	$wpdb->query( "DELETE meta FROM $wpdb->postmeta AS meta LEFT JOIN $wpdb->posts AS posts ON posts.ID = meta.post_id WHERE posts.ID IS NULL;" );
```

Comment:
As below, I think we should delete all the specific postmeta from posts that are specifically of the `amp_validated_url` post type.

The `WHERE posts.ID IS NULL` here makes me nervous, as it seems to rely on the foreign key constraints in InnoDB. If a database is using MyISAM, I don't believe `ID` would ever end up getting nullified here after the posts are deleted. MyISAM doesn't support `FOREIGN KEY` constraints. 

So in order to ensure we're compatible with all MySQL database engines (like MyISAM) or other SQL databases entirely (like SQLite, although I think it supports `FOREIGN KEY` now).

I think WooCommerce may be presuming InnoDB. While I think this is by far the most common engine used in WordPress, it's probably safer to not rely on any `FOREIGN KEY` constraints being present.

Reply:
So I think this `DELETE postmeta` statement should come _before_ the `$wpdb->delete( $wpdb->posts, … )` call, and then `DELETE postmeta` should delete the postmeta specifically related to `amp_validated_url` posts.